### PR TITLE
feat: add release please action for mesosphere branch

### PR DESCRIPTION
**Pull request by mhrabovcin**

### CODE REVIEW

File: .github/workflows/release-please-mesosphere.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  actions: write
+
+jobs:
+  release-please:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: googleapis/release-please-action@v4
+        id: release-please
+        with:
+          release-type: go
```
Comment (jimmidyson) [inline .github/workflows/release-please-mesosphere.yaml]: You should switch to manifest release type instead.

Comment (mhrabovcin) [reply .github/workflows/release-please-mesosphere.yaml]: Fixed.

File: release-please-config.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  "packages": {
+    ".": {
+      "changelog-path": "CHANGELOG.md",
+      "release-type": "go",
+      "skip-github-release": true,
```
Comment (jimmidyson) [inline release-please-config.json]: You don't want a GH release to be created?

Comment (jimmidyson) [reply release-please-config.json]: See https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/blob/main/release-please-config.json for an example of what we do (including updating extra files, but you might not need to do that).

Comment (mhrabovcin) [reply release-please-config.json]: I've updated config based on the CAREN. Thanks

**Review by jimmidyson — APPROVED** [review]

**Pull request merged**